modules/tensor_mechanics/include/materials/ComputeReturnMappingStress.h

```diff
@@ -47,12 +48,18 @@ class ComputeReturnMappingStress : public ComputeFiniteStrainElasticStress
   const bool _output_iteration_info;
   ///@}
 
+  /// after updateQpStress, rotate the stress, elastic_strain, plastic_strain and Jacobian_mult using _rotation_increment
+  const bool _perform_finite_strain_rotations;
+
   ///@{ Rank-4 and Rank-2 elasticity and elastic strain tensors
   const MaterialProperty<RankFourTensor> & _elasticity_tensor;
   MaterialProperty<RankTwoTensor> & _elastic_strain_old;
   const MaterialProperty<RankTwoTensor> & _strain_increment;
   ///@}
 
+  /// plastic strain
+  MaterialProperty<RankTwoTensor> & _plastic_strain;
```

Comment:
Since this base class is used for both creep and plastic models, it should probably have a more generic name like _inelastic_strain.  We were just dealing with a similar issue for the effective plastic strain (or effective creep strain) variable, and we're still working out a solution.  I like the idea of the base class registering the property, but having the derived class tell the base class what the name of it is.